session/session.go

```diff
@@ -540,6 +541,22 @@ func (s *session) CommitTxn(ctx context.Context) error {
 
 	var commitDetail *execdetails.CommitDetails
 	ctx = context.WithValue(ctx, execdetails.CommitDetailCtxKey, &commitDetail)
+	is := infoschema.GetInfoSchema(s)
+	txnScope := s.GetSessionVars().TxnScope
+	for physicalTableID := range s.sessionVars.TxnCtx.TableDeltaMap {
+		bundle, ok := is.RuleBundles()[placement.GroupID(physicalTableID)]
```

Comment:
Use `BundleByName`.

Reply:
A lock will be used in `BundleByName` in the future. It may be concurrently written right now.